src/cluster.ts

```diff
@@ -412,23 +414,9 @@ Please use the format 'my-cluster' or '${instance.name}/clusters/my-cluster'.`);
 
     const reqOpts: ICluster = {
       name: this.name,
+      serveNodes: metadata.nodes,
     };
 
-    if (metadata.location) {
-      reqOpts.location = Cluster.getLocation_(
-        this.bigtable.projectId,
-        metadata.location
-      );
-    }
-
-    if (metadata.nodes) {
-      reqOpts.serveNodes = metadata.nodes;
-    }
-
-    if (metadata.storage) {
-      reqOpts.defaultStorageType = Cluster.getStorageType_(metadata.storage);
-    }
-
```

Comment:
`storage` and `location` can not be updated on an existing cluster

Reply:
I haven't tested this or anything, but the docs say that the UpdateCluster rpc accepts a full "Cluster" object: https://cloud.google.com/bigtable/docs/reference/admin/rpc/google.bigtable.admin.v2#google.bigtable.admin.v2.BigtableInstanceAdmin.UpdateCluster-table

Will it return an error if certain fields are attempted to be updated?

Reply:
It won't return an error,
This change is done to eliminate a possibility of a confusion from the user side if attempted to change a filed that is not supported.

Reply:
I see, thanks. I'm not sure we should use the whitelist approach of only allowing "serveNodes", when the API could evolve someday to accept something else, and we'd be blocking it. Does the upstream team know their API docs state multiple unsupported properties can be sent during the UpdateCluster call, and maybe they would want to make it return an error from their end, or some other solution?

We could also throw from our library if we receive unsupported input, but in general, we're best to simply match what the official docs state is supported. As soon as we go off that path, things get shaky.

Reply:
> Does the upstream team know their API docs state multiple unsupported properties can be sent during the UpdateCluster call

While UpdateCluster rpc does accept a full "Cluster" object, it seems that upstream API docs do specify that among `Cluster` properties

- `name` is an _OutputOnly_ field
- `state` is an _OutputOnly_ field as well
- `location` is a _CreateOnly_ field
- `default_storage_type` is a _CreateOnly_ field as well
- leaving only `serveNodes ` field that can be updated

>  when the API could evolve someday to accept something else, and we'd be blocking it

If/when upstream API evolves to accept something else, that logic could be added as a new feature PR?

WDYT?

Reply:
> If/when upstream API evolves to accept something else, that logic could be added as a new feature PR?

There's usually a big lag before we notice.

If the upstream API silently drops properties, I would think our best bet is to do the same. In terms of code, let's pass them all through. In terms of types, maybe that's where you could just list "serveNodes"? At least then, if the user knows better and is trying to set a property we haven't added to the types yet, they could have their TypeScript compiler ignore that (I think?).

Reply:
Added the logic back, left the types with `nodes` only.

Reply:
@stephenplusplus WDYT?